docs/adr/0002-use-react-redux-and-typescript-for-subscription-management-pages.md

```diff
@@ -0,0 +1,129 @@
+
+## Decision Drivers
+
+* Opportunity for a fresh start with tech stack without rewriting
+* Security in dealing with payment transactions
+* Developer ergonomics
+* Code quality & testing
+* Subscription services deadlines
+
+## Considered Options
+
+* React, Redux, Typescript
+* Backbone.js
+
+## Decision Outcome
+
+Chosen option: "React, Redux, Typescript", because
+
+* Chance to start with a fresh stack
+* More vibrant ecosystem
+* Better tooling & developer ergonomics
+
+## Pros and Cons of the Options
+
+### React, Redux, TypeScript
+
+[React][] is a library for building DOM-based UI components in a data-driven,
+declarative style that minimizes granular DOM manipulation by developers by way
+of a Virtual DOM diff / patch algorithm. It also introduces patterns & tools for
+encapsulating reusable UI functionality in "hooks" and JS classes.
+
+[React]: https://reactjs.org/
+
+[Redux][] is a library for managing app state in a singleton container using a
+defined inventory of actions that describe changes to be applied via reducer
+functions and middleware filters.
+
+[Redux]: https://redux.js.org/
+
+[TypeScript][] is a superset of JavaScript which introduces type annotation
+expressions and type inference along with tooling to lint and compile.
+
+[TypeScript]: https://www.typescriptlang.org/
+
+We can also consider other dependencies that act as glue between these pieces
+and utilities to reduce boilerplate and repetition. (e.g. [redux-actions][] to
+simplify constructing actions; [redux-promise-middleware][] for actions that
+result in async API fetches; [redux-thunk][] for producing action sequences)
+
+[redux-actions]: https://redux-actions.js.org/ 
+
+[redux-promise-middleware]:
+https://github.com/pburtchaell/redux-promise-middleware 
+
+[redux-thunk]: https://github.com/reduxjs/redux-thunk
+
+* Pros
+  * Frequently updated and vibrant ecosystem
+  * New team members may be more familiar with the React / Redux / TypeScript
+    ecosystem
+  * React integrates well with Webpack for code-splitting and lazy-loaded
+    component JS bundles
+  * React and Webpack can support per-component hot reloading rather than
+    reloading the whole page
+  * React can organize UI rendering into small components with single
+    responsibility for each
+  * React hooks can be used to encapsulate reusable chunks of UI logic & state
+    between components
+  * Redux can organize business logic and API requests in a central set of
+    modules
+  * Data-driven React components can be loosely-coupled with Redux state
+  * The Redux dev tools browser add-on can monitor, inspect, and even rewind app
+    state changes
+  * TypeScript can help document & validate data structures at build time (e.g.
+    in Redux state) and integrate with IDEs (e.g. intellisense hints in VS Code)
+  * Can follow the Mocha / Chai / Sinon style unit testing stack used elsewhere
+    in the project
+* Cons
```

Comment:
One thing that should be mentioned here is that we haven't figured out the l10n story for React/Redux and this project yet.

Reply:
I think we're punting L10n for the MVP of subscriptions. But, I think we were thinking of using [Fluent's React bindings](https://github.com/projectfluent/fluent.js/wiki/React-Bindings) and FTL in Pontoon. I can add that to the doc